### PR TITLE
New EF build image

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Please do not merge yet. I would like to do few experiments with jobs.

### CODE REVIEW

File: etc/jenkins/docker/image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Distribution License v. 1.0, which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (lukasj) [inline etc/jenkins/docker/image/Dockerfile:7]: project is under `SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause`

File: etc/jenkins/docker/pom.xml [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    <artifactId>org.eclipse.persistence.build.docker</artifactId>
+    <!-- Version is used as docker image tag -->
+    <version>2.0.0-EA1</version>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <docker.image>tkraus/el-build</docker.image>
+        <docker.source>oraclelinux:8</docker.source>
+        <java.pkg>openjdk-17.0.2_linux-x64_bin.tar.gz</java.pkg>
+        <java.url>https://download.java.net/java/GA/jdk17.0.2/dfd4a8d0985749f896bed50d7138ee7f/8/GPL/${java.pkg}</java.url>
```
Comment (lukasj) [inline etc/jenkins/docker/pom.xml:29]: What is this? Why not to use https://www.oracle.com/java/technologies/jdk-script-friendly-urls/ ?

File: etc/jenkins/docker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+                    <execution>
+                        <id>clean-image</id>
+                        <phase>clean</phase>
+                        <goals>
+                            <goal>remove</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- Disable resources -->
```
Comment (lukasj) [inline etc/jenkins/docker/pom.xml:218]: doable via `<maven.resources.skip>true</maven.resources.skip>` property

File: etc/jenkins/docker/pom.xml [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-testResources</id>
+                        <phase>none</phase>
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- Disable compiler -->
```
Comment (lukasj) [inline etc/jenkins/docker/pom.xml:229]: why not:
```
<maven.main.skip>true</maven.main.skip>
<maven.test.skip>true</maven.test.skip>
```
instead?

File: etc/jenkins/docker/pom.xml (hunk 3 above)
Comment (lukasj) [reply etc/jenkins/docker/pom.xml:218]: should use `pom` packaging type then

File: etc/jenkins/docker/image/Dockerfile (hunk 1 above)
Comment (Tomas-Kraus) [reply etc/jenkins/docker/image/Dockerfile]: fixed.

File: etc/jenkins/docker/pom.xml (hunk 2 above)
Comment (Tomas-Kraus) [reply etc/jenkins/docker/pom.xml]: fixed.

File: etc/jenkins/docker/pom.xml (hunk 3 above)
Comment (Tomas-Kraus) [reply etc/jenkins/docker/pom.xml]: ok, changed to pom packaging.

File: etc/jenkins/docker/pom.xml (hunk 4 above)
Comment (Tomas-Kraus) [reply etc/jenkins/docker/pom.xml]: I had to rewrite everything here after changing packaging to pom. Now rest of the plugins can be resolved using properties.

**Review by rfelcman — APPROVED** [review]
LGTM

**Pull request merged**